vault/namespace_store.go

```diff
@@ -510,17 +526,18 @@ func (ns *NamespaceStore) GetNamespaceByLongestPrefix(ctx context.Context, path
 func (ns *NamespaceStore) GetNamespaceByPath(ctx context.Context, path string) (*namespace.Namespace, error) {
 	defer metrics.MeasureSince([]string{"namespace", "get_namespace_by_path"}, time.Now())
 
-	if err := ns.checkInvalidation(ctx); err != nil {
+	unlock, err := ns.lockWithInvalidation(ctx, false)
+	if err != nil {
 		return nil, err
 	}
+	defer unlock()
 
-	ns.lock.RLock()
-	defer ns.lock.RUnlock()
-
-	return ns.getNamespaceByPathLocked(ctx, path)
+	return ns.getNamespaceByPathLocked(ctx, path, false)
 }
 
-func (ns *NamespaceStore) getNamespaceByPathLocked(ctx context.Context, path string) (*namespace.Namespace, error) {
+func (ns *NamespaceStore) getNamespaceByPathLocked(
+	ctx context.Context, path string, withUnlock bool,
```

Comment:
```suggestion
	ctx context.Context, path string, withUnlockKey bool,
```

Stating `withUnlock`, especially in a PR about mutex locking, makes it seem like this should call unlock(...) somewhere. :-D